tests/test_read_pdf_table.py

```diff
@@ -30,6 +30,11 @@ def test_read_remote_pdf(self):
         df = tabula.read_pdf(uri)
         self.assertTrue(isinstance(df, pd.DataFrame))
 
+    def test_read_remote_pdf_with_custom_user_agent(self):
+        uri = "https://github.com/tabulapdf/tabula-java/raw/master/src/test/resources/technology/tabula/12s0324.pdf"
+        df = tabula.read_pdf(uri, user_agent='Mozilla/5.0')
+        self.assertTrue(isinstance(df, pd.DataFrame))
```

Comment:
It seems this test doesn't check user agent. Would it be possible to check it?

Reply:
I'm trying to figure out a way to test it. Can I create a local server using flask for this purpose. Or do you have some better ideia?

Reply:
IMHO, That's too heavy for this test.

I thought Mocking and validating `urlopen` with the expected `Request` object can be fine for it. https://docs.python.org/3/library/unittest.mock.html#quick-guide 
How about this approach? Or, if it doesn't work for it, it's ok to keep as is.

Reply:
I tried this without success. I create a new function with the functionality to create the request and call it on the tests

Reply:
I mean I think you can write a test with `patch` like this https://github.com/chezou/tabula-py/blob/master/tests/test_read_pdf_table.py#L182-L191

It's OK to create a new function but `assert_called_with` is enough in this case.

Reply:
I tried here. Didn't work. I'll try to figure out still this week how to test with the `assert_called_with`